doc/reference/tt_cli/external_modules.rst

```diff
@@ -0,0 +1,71 @@
+Extending the tt functionality
+==============================
+
+The ``tt`` utility implements the modular architecture: its :doc:`commands <commands>`
+are implemented in separate modules. When you run ``tt`` with a command, the
+corresponding module is executed with the given arguments.
+
+Modular architecture enables the option to extend the ``tt`` functionality with
+**external modules** (as opposed to **internal modules** that implement built-in
+commands). Simply said, you can write any code you want to execute
+from ``tt``, pack it into an executable, and run it with a ``tt`` command:
+
+..  code-block:: bash
+
+    tt my-module-name my-args
+
+The command that executes a module is the same as the name of its executable.
+
+
+Module description and help
+---------------------------
+
+Executables that implement external ``tt`` modules must have two flags:
+
+-   ``--description`` -- print a short description of the module. It is shown alongside
+    the command in the ``tt`` help.
+-   ``--help`` -- display help. It is shown when ``tt <module-name> help`` is called.
```

Comment:
Actually, yes, but no) We pass all options and arguments that were specified by the user into the module (and `tt EXT_MODULE_NAME --help` will work fine), but the `help` module will also output help from the `tt help EXT_MODULE_NAME` external module if `--help` is implemented in an external module.